stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/model/ChannelConfig.kt

```diff
@@ -0,0 +1,11 @@
+package io.getstream.chat.android.livedata.model
+
+import io.getstream.chat.android.client.models.Config
+
+internal typealias ChannelConfig = Pair<String, Config>
+
+internal val ChannelConfig.type: String
+    get() = this.first
+
+internal val ChannelConfig.config: Config
+    get() = this.second
```

Comment:
What's the benefit of this over a simple data class?

Reply:
Why do you need to create any new class if it's just a simple pair? It covers all required use-cases for ChannelConfig.

Reply:
For the exact reason why you've introduced the two extension properties here, to give the two values in it meaningful names. Creating new classes is super cheap in Kotlin - `Pair` really only exists to make creating Maps easier with `mapOf` (no idea why `Triple` exists). If you need a tuple, you can always create a meaningful data class in a single line for it.

With this solution, we have more code, we have `first` and `second` as available properties on `ChannelConfig`, as well as in its `toString` implementation, and other Pairs of these two values are cross-assignable with `ChannelConfig`.

Reply:
I don't have any strong point against implementation via Pair. From my point of view, it's the same, and data classes contain their `componentN()` functions that can confuse (but really no one gets). Feels like a matter of taste. But if you really don't like it. I can change easily

P.S. `Pair`, `Triple` are famous Tuple concept. And it's one of the weakest parts of the Kotlin language design. For example [C# has a much better implementation](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/value-tuples)

Reply:
`Pair` is also a data class, so it'll also have the component functions and all that functionality. The only difference between using `Pair` and your own data class is that you get to name your values nicely. Makes them easier to access, gives you a correct `toString` implementation, correct parameter names in the `copy` method, etc.

Kotlin used to have tuples, but since data classes are so concise and easy to create wherever you need them, they removed separate support for them. [This article](https://blog.jetbrains.com/kotlin/2012/09/how-do-you-traverse-a-map/) shows off some ancient history for this 😄

Reply:
About topic - I'll change to data classes.

About tuples. I think it's also defined by the support requirement of java SDK background and its' work with `Map` and so on. data classes are too wordy when you want to just receive some values as `(Long, String, Double, SomeOtherType)`. It couldn't be related to logic/domain entoty model